### ISSUE TITLE
Wrap every component in an application wrapper

### ISSUE THREAD

**Issue by benbayard** [happo/happo.io#25]
hey folks! I was wondering if there was a way to wrap every happo example with another component. 

Since we use `styled-components` + a theme we always wrap our happo examples in a theme provider: 

```
export const medium = () => (
    <ThemeProvider>
        <MyComponent... />
    </ThemeProvider>
)
```

There are ways we can obfuscate this away ourselves, for instance:

```
export const medium = provideTheme(() => ...)
```

But I'd rather be able to configure happo to do this for us, so that all our components are wrapped behind the scenes.

Is this possible now? If not, I am open to taking up the OSS work required to make it work unless you have plans / API docs ready now.

**Comment by trotzig** [comment]
Hi 👋

Right now this isn't possible. It has come up before however, and I've thought about doing it myself but never got around to it. 

If you wanted to give it an attempt, this is where we'd have to inject the wrapper: https://github.com/enduire/happo.io/blob/d6fb633c3e4ba6bc5483938b7b7c8ffccecbbe48/src/createDynamicEntryPoint.js#L36

Things are made a little complicated due to the fact that webpack is involved here, and that particular line is injected as a string. One way to implement this would be to add a configuration option for a path to a file that we assume export a wrapper function. Something like

```js
// .happo.js
module.exports = {
  renderWrapperModule: path.resolve(__dirname, 'happo-wrapper.js'), 
}
```
And then we'd require that module in our webpack bundle, calling the wrapper method as part of rendering.

**Comment by benbayard** [comment]
Yup, that's exactly what I was thinking! Thank you for the starting point, I'll talk to my team about this today!

**Comment by trotzig** [comment]
Another idea would be to leverage the already-existing `setupScript` config option. Either use what’s exported from that file/module or allow people to define a global `window.wrapRender` function.

**Comment by trotzig** [comment]
@benbayard I pushed a change with support for wrapping, it's been released in 3.1.0. Let me know how things work out! Here's a link to docs: 
https://github.com/happo/happo.io#renderwrappermodule
